src/Platform/Microsoft.Testing.Platform/TestHostOrcherstrator/ITestHostOrchestratorApplicationLifetime.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Testing.Platform.Extensions.TestHostOrchestrator;
+
+// NOTE: The equivalent of this for "test host" is ITestApplicationLifecycleCallbacks, which is an unfortunate naming.
```

Comment:
We could ship one for TestHostControllers, one for TestHost (with similar naming pattern), and deprecate the current `ITestApplicationLifecycleCallbacks` (suggesting to use `ITestHostApplicationLifetime` instead). That would put us in a better shape for v2.